zulipterminal/model.py

```diff
@@ -996,11 +1000,39 @@ def get_stream_by_id(streams: List[StreamData], stream_id: int) -> StreamData:
                         for user_id in user_ids:
                             subscribers.remove(user_id)
 
+    @asynch
+    def _show_typing_animation(self, recipient: Dict[str, Any]) -> None:
```

Comment:
This isn't just showing a typing animation, it's now the entire typing notification function.

Reply:
Yes one could say that, but it is mainly updating the animation "frame" by "frame" and deciding when to end it.
If future animations are done then  `create_animation` or similar function could be extracted out of this. For now I think calling it show_typing_notification and moving it to Controller would be better.

Reply:
Your suggestion was along the lines of what I was meaning :+1: